unity/unity-embed-host/Extensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Unity.CoreCLRHelpers;
+
+static class Extensions
+{
+    public static Type TypeFromHandleIntPtr(this nint intPtrToTypeHandle)
+        => Type.GetTypeFromHandle(RuntimeTypeHandle.FromIntPtr(intPtrToTypeHandle));
+
+    public static nint TypeHandleIntPtr(this object obj)
+        => obj.GetType().TypeHandleIntPtr();
+
+    public static nint TypeHandleIntPtr(this Type type)
+        => RuntimeTypeHandle.ToIntPtr(type.TypeHandle);
+
+    public static T UnsafeAs<T>(this nint intPtr)
```

Comment:
Nit, the extension below is clearly to convert an object to an IntPtr/nint representation in the embedding API. I'm not sure of the best way of organizing it, but we want to make sure we keep all object<->nint and nint<->object conversions in one place (and only use methods for that) because at some point that will switch from just an unsafe cast to allocating/retrieving GC Handles.

Reply:
Basically, any objects (reference types) being passes as parameters or as a return value on the embedding interface need special handling.

Reply:
That's my initial thought. I don't think it's required, just want to ensure we keep those specific operations isolated to we can change them easily without having to re-audit all the code.